apps/report/views.py

```diff
@@ -60,10 +61,18 @@ def get_context_data(self, **kwargs):
             "group__name").annotate(dcount=Count("group")).order_by("-dcount")
         context["repotest"] = Test.objects.values("git__name").annotate(
             dcount=Count("git")).order_by("-dcount")
-        context["testlengths"] = Task.objects.filter(
-            datestart__gte=datetime.now()+timedelta(hours=-168)).values(
-            "test__name").annotate(avg_duration=Avg('duration')).order_by(
-            "-avg_duration")
+        testlengths = \
+            Task.objects.filter(recipe__job__schedule__date_create__gte=datetime.now() +
+                                timedelta(hours=-settings
+                                .LONGEST_RUNNING_PERIOD))\
+            .values("test__name").annotate(avg_duration=Avg('duration'))\
+            .annotate(runs=Count('test_id'))\
+            .order_by("-avg_duration")
+        context["testlengths1"] = \
+            testlengths[:settings.LONGEST_RUNNING_COLUMN_LENGTH]
```

Comment:
not divide in code, divide in template by call "slice" https://docs.djangoproject.com/en/1.9/ref/templates/builtins/#slice